File: go/test/protocol/test_case_cloud_event.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type testCaseCloudEvent struct {
	Source          *string `yaml:"source"`
	Type            *string `yaml:"type"`
	SpecVersion     *string `yaml:"spec-version"`
	ID              *string `yaml:"id"`
	Time            any     `yaml:"time"`
	DataContentType *string `yaml:"data-content-type"`
	Subject         any     `yaml:"subject"`
	DataSchema      any     `yaml:"data-schema"`
}

type TestCaseCloudEvent struct {
	testCaseCloudEvent
}

func (cloudEvent *TestCaseCloudEvent) UnmarshalYAML(
	node *yaml.Node,
) error {
	*cloudEvent = TestCaseCloudEvent{}

	cloudEvent.Time = false
	cloudEvent.Subject = false
	cloudEvent.DataSchema = false

	return node.Decode(&cloudEvent.testCaseCloudEvent)
}

File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseCloudEvent.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseCloudEvent
    {
        internal bool Irrelevant { get; set; } = false;

        public string? Source { get; set; }

        public string? Type { get; set; }

        public string? SpecVersion { get; set; }

        public string? Id { get; set; }

        public object Time { get; set; } = false;

        public string? DataContentType { get; set; }

        public object Subject { get; set; } = false;

        public object DataSchema { get; set; } = false;
    }
}
